### PR TITLE
Guarantee deps are built before finishing import.

### PR DESCRIPTION
**Pull request by andrefmrocha**
Dependencies must be fully built in order to guarantee the dependencies are all present for indexing.

Opening this one as a PR since I'm not sure if there's a better way of doing this.

### CODE REVIEW

File: src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java [hunk 1]
```diff
@@ -146,6 +146,33 @@ private BuildTarget getBuildTargetForRule(Build.Rule rule) {
         .collect(Collectors.toList());
   }
 
+  public List<BuildTargetIdentifier> getTargetIdentifiersForDependencies(
+      List<BuildTargetIdentifier> targets) {
+    String targetsUnion =
```
Comment (Abrams27) [inline src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java:151]: It looks like a good thing to move to the `TargetsUtils`

File: src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspCompilationManager.java [hunk 2]
```diff
@@ -49,16 +49,19 @@ public BazelBspCompilationManager(BazelRunner bazelRunner, BazelData bazelData)
...
-      target.getRule().getRuleOutputList().stream()
-          .filter(output -> output.contains(Constants.DIAGNOSTICS))
-          .forEach(
-              output ->
-                  diagnosticsProtosLocations.put(
-                      target.getRule().getName(),
-                      BuildManagerParsingUtils.convertOutputToPath(
-                          output, bazelData.getBinRoot())));
-    }
+    queryResult.getTargetList().stream()
```
Comment (Abrams27) [inline src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspCompilationManager.java:52]: this sequence becomes long and complex, maybe it should be splitted somehow

**Review by Abrams27 — COMMENTED** [review]
hopefully it solves the problem, left minor comments

File: src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java (hunk 1 above)
Comment (andrefmrocha) [reply src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java:151]: the whole method or just getting the union? I'm not sure what you mean from your selection  :sweat_smile:

Comment (Abrams27) [reply src/main/java/org/jetbrains/bsp/bazel/server/bsp/managers/BazelBspQueryManager.java:151]: this part: `targets.stream().map(BuildTargetIdentifier::getUri).collect(Collectors.joining("+"));`

**Pull request merged**